stan/math/prim/prob/cauchy_lpdf.hpp

```diff
@@ -53,57 +58,76 @@ return_type_t<T_y, T_loc, T_scale> cauchy_lpdf(const T_y& y, const T_loc& mu,
   }
 
   T_partials_return logp(0.0);
-  operands_and_partials<T_y, T_loc, T_scale> ops_partials(y, mu, sigma);
+  operands_and_partials<T_y_ref, T_mu_ref, T_sigma_ref> ops_partials(
+      y_ref, mu_ref, sigma_ref);
 
-  scalar_seq_view<T_y> y_vec(y);
-  scalar_seq_view<T_loc> mu_vec(mu);
-  scalar_seq_view<T_scale> sigma_vec(sigma);
-  size_t N = max_size(y, mu, sigma);
+  const auto& y_col = as_column_vector_or_scalar(y_ref);
+  const auto& mu_col = as_column_vector_or_scalar(mu_ref);
+  const auto& sigma_col = as_column_vector_or_scalar(sigma_ref);
 
-  VectorBuilder<true, T_partials_return, T_scale> inv_sigma(size(sigma));
-  VectorBuilder<true, T_partials_return, T_scale> sigma_squared(size(sigma));
-  VectorBuilder<include_summand<propto, T_scale>::value, T_partials_return,
-                T_scale>
-      log_sigma(size(sigma));
-  for (size_t i = 0; i < stan::math::size(sigma); i++) {
-    const T_partials_return sigma_dbl = value_of(sigma_vec[i]);
-    inv_sigma[i] = 1.0 / sigma_dbl;
-    sigma_squared[i] = sigma_dbl * sigma_dbl;
-    if (include_summand<propto, T_scale>::value) {
-      log_sigma[i] = log(sigma_dbl);
-    }
-  }
+  const auto& y_arr = as_array_or_scalar(y_col);
```

Comment:
Why does this need to go `as_column_vector_or_scalar` to `as_array_or_scalar`?

Can it not go directly `as_array_or_scalar`?

Reply:
Distributions can accept combination of row and column vector inputs. To make dimensions match when doing math we are transposing any row vectors.

Reply:
Right, but that's what `as_column_vector_or_scalar` does.

So we could just do this and not worry about the arrays:
```cpp
const auto& y_col = as_column_vector_or_scalar(y_ref);
const auto& mu_col = as_column_vector_or_scalar(mu_ref);
const auto& sigma_col = as_column_vector_or_scalar(sigma_ref);

ref_type_t<decltype(value_of(y_arr))> y_val = value_of(y_col);
ref_type_t<decltype(value_of(mu_arr))> mu_val = value_of(mu_col);
ref_type_t<decltype(value_of(sigma_arr))> sigma_val = value_of(sigma_col);
```

Reply:
Arrays are necesary to do things such as addition of an array and a scalar. Such operation is not defined for matrices. 

It is true, however that these two functions are usually called together. So we could make a single function that does both of these.

Reply:
Oh that's array in the Eigen sense, not the Stan sense. Given that I definitely think it makes sense that `as_column_vector_or_scalar` and `as_array_or_scalar` should take in the same types (since the difference in a column vector an an array is a `.array()`), but that can be a different pull or whatever. This is good.